internal/adapters/entrypoints/rest/common.go

```diff
@@ -24,17 +27,43 @@ var RequestValidator = validator.New(validator.WithRequiredStructEnabled())
 
 func PositiveStringValidationRule(value string) bool {
 	bigIntValue := new(big.Int)
-	bigIntValue.SetString(value, 10)
+	_, ok := bigIntValue.SetString(value, 10)
+	if !ok {
+		return false
+	}
 	return bigIntValue.Cmp(big.NewInt(0)) > 0
 }
 
+func decimalPlacesValidator(fl validator.FieldLevel) bool {
```

Comment:
I'd recommend to test this validator too, because the implementation is not straight forward and we want to make sure that 1e-9 is enough in case the floating point operations result in more decimals than expected